github/GithubIntegration.py

```diff
@@ -212,18 +218,29 @@ def create_jwt(self, expiration: int | None = None) -> str:
         return self.auth.create_jwt(expiration)
 
     def get_access_token(
-        self, installation_id: int, permissions: dict[str, str] | None = None
+        self,
+        installation_id: int,
+        permissions: dict[str, str] | None = None,
+        repositories: list[str] | None = None,
+        repository_ids: list[int] | None = None,
     ) -> InstallationAuthorization:
         """
         :calls: `POST /app/installations/{installation_id}/access_tokens <https://docs.github.com/en/rest/apps/apps#create-an-installation-access-token-for-an-app>`
         """
+
         if permissions is None:
             permissions = {}
 
+        if repositories is not None and not isinstance(repositories, list):
+            raise GithubException(status=400, data={"message": "Invalid repositories"}, headers=None)
+
+        if repository_ids is not None and not isinstance(repository_ids, list):
+            raise GithubException(status=400, data={"message": "Invalid repository_ids"}, headers=None)
+
         if not isinstance(permissions, dict):
             raise GithubException(status=400, data={"message": "Invalid permissions"}, headers=None)
 
-        body = {"permissions": permissions}
+        body = {"permissions": permissions, "repositories": repositories, "repository_ids": repository_ids}
```

Comment:
This will remove the elements that are `None`:
```suggestion
        body = {"permissions": permissions}
        if repositories is not None:
          body["repositories"] = repositories
        if repository_ids is not None:
          body["repository_ids"] = repository_ids
```